keps/sig-node/4569-cgroup-v1-maintenance-mode/README.md

```diff
@@ -0,0 +1,652 @@
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+1. Monitor existing cgroup v2 jobs, e.g.
+  - https://testgrid.k8s.io/sig-node-release-blocking#ci-crio-cgroupv2-node-e2e-conformance
+
+2. Ensure all features coverage by running all tests on cgroup v2 (while some may still run on cgroup v1 to test back compatibility)
+
+3. Make cgroup v2 host mandatory for new e2e and node e2e tests.
+
+### Graduation Criteria
+
+#### Alpha
+
+N/A
+
+#### Beta
+
+N/A
+
+#### GA
+- Kubelet detects the host using cgroup v1, it will not only log a warning message but also generate an event to highlight the cgroup v1 moving to maintenance mode.
+- Introduce a new metric, `kubelet_cgroup_version`, to provide insights into the cgroup version utilized by the hosts.
+- Introduce a boolean kubelet flag `--disable-cgroupv1-support` and set it to `false` by default.
+- Blog post on advantages of using cgroup v2 with kubernetes.
+- Code modifications for to assume cgroup v2 by default. Check [this]( #code-modifications-for-default-cgroup-assumptions) section for details.
+- Ensure all features coverage by running all tests on cgroup v2 (while some may still run on cgroup v1 to test back compatibility)
+- Make cgroup v2 host mandatory for new e2e and node e2e jobs. Set `--disable-cgroupv1-support` to `true` for those jobs.
+- Fix all pending known bugs in cgroup v2 support in kubernetes.
+- Separation of cgroup v1 and cgroup v2 Code Paths. Check [this](#separation-of-cgroup-v1-and-cgroup-v2-code-paths) section for details.
+
+- Mark cgroup v1 support in maintenance mode in the [documenatation](https://kubernetes.io/docs/home/).
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+- For clusters upgrading to a version of Kubernetes where cgroup v1 is in maintenance mode, administrators should ensure that all nodes are compatible with cgroup v2 prior to upgrading. This might include operating system upgrades or workload configuration changes.
+- Downgrading to a version that supports cgroup v1 should not require any special considerations regarding cgroup version, as cgroup v2 is backwards compatible with cgroup v1 from a Kubernetes perspective. However, specific node configurations or workload resource requirements may necessitate adjustments.
```

Comment:
This isn't quite true as with the current plans we may have cgroup-v2-only default-on functionality that is lost when downgrading?

We should probably recommend that people migrate to cgroup v2 _before_ upgrading Kubernetes? (and they can roll back the kernel/host change)

Reply:
That's a good point. Some kubelet features like swap support only work with v2, so we need to call that out. Thanks for pointing that out. I will update that text.